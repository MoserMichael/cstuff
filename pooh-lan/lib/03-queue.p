include 'queue.inc'

println( ~msg 'queue of numbers' )

l = make_queue()

l . push( ~data 1 )
l . push( ~data 2 )
l . push( ~data 3 )
l . push( ~data 4 )

while  l . count() != 0
  r = l . pop( )
  println( ~msg r )
end

println( ~msg 'stack of arrays' )

l . push( ~data [ 1, 2] )
l . push( ~data [ 2, 3] )
l . push( ~data [ 3, 4] )
l . push( ~data [ 5, 6] )

while  l . count() != 0
  r = l . pop( )
  println( ~msg join( ~array r ~separator ' - ' ) )
end



print( ~msg '*** eof stack test ***' )



