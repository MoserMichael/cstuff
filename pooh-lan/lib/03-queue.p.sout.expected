queue of numbers
1
2
3
4
stack of arrays
1 - 2
2 - 3
3 - 4
5 - 6
*** eof stack test ***